sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -0,0 +1,218 @@
+    public class TestProxy
+    {
+        private static bool HasDotNetExe(string dotnetDir) => dotnetDir != null && File.Exists(Path.Combine(dotnetDir, s_dotNetExeName));
+
+        private static readonly string s_dotNetExeName;
+
+        private static readonly string s_dotNetExe;
+
+        public const string DevCertIssuer = "CN=localhost";
+
+        public const string IpAddress = "127.0.0.1";
+
+        public int? ProxyPortHttp => _proxyPortHttp;
+
+        public int? ProxyPortHttps => _proxyPortHttps;
+
+        private readonly int? _proxyPortHttp;
+        private readonly int? _proxyPortHttps;
+        private readonly Process _testProxyProcess;
+        internal TestProxyRestClient Client { get; }
+
+        static TestProxy()
+        {
+            string installDir = Environment.GetEnvironmentVariable("DOTNET_INSTALL_DIR");
+            s_dotNetExeName = "dotnet" + (Path.DirectorySeparatorChar == '/' ? "" : ".exe");
+            if (!HasDotNetExe(installDir))
+            {
+                installDir = Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator).FirstOrDefault(HasDotNetExe);
+            }
+
+            if (installDir == null)
+            {
+                throw new InvalidOperationException("DOTNET install directory was not found");
+            }
+
+            s_dotNetExe = Path.Combine(installDir, s_dotNetExeName);
+        }
+
+        private TestProxy(string proxyPath)
+        {
+            ProcessStartInfo testProxyProcessInfo = new ProcessStartInfo(
+                s_dotNetExe,
+                proxyPath)
+            {
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                EnvironmentVariables =
+                {
+                    ["ASPNETCORE_URLS"] = $"http://{IpAddress}:0;https://{IpAddress}:0",
+                    ["Logging__LogLevel__Microsoft"] = "Information"
+                }
+            };
+
+            if (!TestEnvironment.GlobalIsRunningInCI)
+            {
+                ImportDevCertIfNeeded();
+            }
+
+            _testProxyProcess = Process.Start(testProxyProcessInfo);
+            ProcessTracker.Add(_testProxyProcess);
+            _ = Task.Run(
+                () =>
+                {
+                    while (!_testProxyProcess.HasExited && !_testProxyProcess.StandardError.EndOfStream)
+                    {
+                        var error = _testProxyProcess.StandardError.ReadLine();
+                        throw new TestProxyException($"An error occurred in the test proxy: {error}");
+                    }
+                });
+
+            int lines = 0;
+            while ((_proxyPortHttp == null || _proxyPortHttps == null) && lines++ < 20)
+            {
+                string outputLine = _testProxyProcess.StandardOutput.ReadLine();
+                // useful for debugging
+                TestContext.Progress.WriteLine(outputLine);
```

Comment:
Are we going to keep this long-term? Doesn't it flood the local run output?

Reply:
It only outputs a few lines (under 5) per test class. I'd like to keep it until we complete the entire migration at least.